File: app/src/lib/types/record.ts
import { Exercise } from './exercise';

export interface Set {
  id: number;
  createdAt: string;
  updatedAt: string;
  reps: number;
  weight: number;
  recordId: number;
}

export interface Record {
  id: number;
  createdAt: string;
  updatedAt: string;
  exerciseId: number;
  exercise: Exercise;
  userId: number;
  sets: Set[];
  date?: string;
}

export interface RecordsResponse {
  records: Record[];
  count: number;
}

export interface CreateSetRequest {
  reps: number;
  weight: number;
}

export interface CreateRecordRequest {
  exerciseId: number;
  sets: CreateSetRequest[];
  date?: string;
}

export interface ExerciseOption {
  id: number;
  name: string;
}

export interface ExerciseOptionsResponse {
  exercises: ExerciseOption[];
  count: number;
} 

File: api/models/record.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Record struct {
	ID        uint           `json:"id" gorm:"primaryKey"`
	CreatedAt time.Time      `json:"createdAt"`
	UpdatedAt time.Time      `json:"updatedAt"`
	DeletedAt gorm.DeletedAt `json:"deletedAt,omitempty" gorm:"index"`

	ExerciseID uint     `json:"exerciseId"`
	Exercise   Exercise `json:"exercise" gorm:"foreignKey:ExerciseID"`
	UserID     uint     `json:"userId"`
	Sets       []Set    `json:"sets"`
	Date       *time.Time `json:"date,omitempty"`
}
